libs/kotaemon/kotaemon/loaders/azureai_document_intelligence_loader.py

```diff
@@ -118,6 +122,8 @@ def load_data(
     ) -> list[Document]:
         """Extract the input file, allowing multi-modal extraction"""
         metadata = extra_info or {}
+        base_name = os.path.basename(file_path)
+        file_name, _ = os.path.splitext(base_name)
```

Comment:
Same with the suggestion above to change `os.path.xxx` to `pathlib`'s equivalent function.